mlir/lib/Dialect/Arith/Utils/Utils.cpp

```diff
@@ -100,6 +100,16 @@ llvm::SmallBitVector mlir::getPositionsOfShapeOne(unsigned rank,
   return dimsToProject;
 }
 
+Value mlir::getValueOrCreateConstantOp(OpBuilder &b, Location loc,
+                                       Type targetType, OpFoldResult ofr) {
+  if (auto value = llvm::dyn_cast_if_present<Value>(ofr))
+    return value;
+  auto attr = dyn_cast<IntegerAttr>(llvm::dyn_cast_if_present<Attribute>(ofr));
```

Comment:
Why do we need two nested casts?

Reply:
Because `OpFoldResult` is a mix of `Value` and `Attribute` and needs conversion to `IntegerAttr`. But you could use `cast<IntergerAttr>` and drop the subsequent assert

Reply:
right, done.